prepare-image.sh

```diff
@@ -41,13 +41,23 @@ if [[ "$INSTALL_TYPE" == "source" ]]; then
 
     python3 -c 'import os; import sys; import jinja2; sys.stdout.write(jinja2.Template(sys.stdin.read()).render(env=os.environ, path=os.path))' < "${IRONIC_PKG_LIST}" > "${IRONIC_PKG_LIST_FINAL}"
 
+    UPPER_CONSTRAINTS_PATH="/tmp/${UPPER_CONSTRAINTS_FILE}"
+
+    # NOTE(elfosardo): if the content of the upper-constraints file is default,
+    # we give as assumed that we're on the master branch
+    if grep -q "this is a placeholder for custom upper-constraints" "${UPPER_CONSTRAINTS_PATH}" ; then
```

Comment:
Ugh. Can we use the file's absence as a sign instead? Or default UPPER_CONSTRAINTS_FILE to an empty string so that COPY also works?

Reply:
Yeah an empty file seems cleaner then you can just do something like `if [ ! -s "${UPPER_CONSTRAINTS_PATH}" ]` here?

Reply:
sounds good to me, I'll make the change
thanks!